bin/check_h5_plugin.cpp

```diff
@@ -1,29 +1,88 @@
 // SPDX-License-Identifier: MIT
 
 #include <dectris/neggia/plugin/H5ToXds.h>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <ios>
 #include <iostream>
 #include <memory>
 
+#include <dlfcn.h>
+
+typedef void (*dl_plugin_open)(const char*,
+                               int info_array[1024],
+                               int* error_flag);
+
+typedef void (*dl_plugin_get_header)(int* nx,
+                                     int* ny,
+                                     int* nbytes,
+                                     float* qx,
+                                     float* qy,
+                                     int* number_of_frames,
+                                     int info_array[1024],
+                                     int* error_flag);
+
+typedef void (*dl_plugin_get_data)(int* frame_number,
+                                   int* nx,
+                                   int* ny,
+                                   int data_array[],
+                                   int info_array[1024],
+                                   int* error_flag);
+
+typedef void (*dl_plugin_close)(int* error_flag);
+
+std::string time_string(std::chrono::system_clock::time_point tp) {
+    std::time_t tp_t = std::chrono::system_clock::to_time_t(tp);
+
+    char s[100];
+    auto n = std::strftime(&s[0], 100, "%Y-%m-%d %H:%M:%S",
+                           std::localtime(&tp_t));
+    return std::string(s, n);
+}
+
 int main(int argc, char* argv[]) {
-    if (argc != 2) {
+    dl_plugin_open open_file;
+    dl_plugin_get_header get_header;
+    dl_plugin_get_data get_data;
+    dl_plugin_close close_file;
+    if (argc < 2) {
```

Comment:
The if-logic here is confusing and wrong. You can fix it and make it more readable with:
```c++
if (argc == 2) {
    // ...
} else if (argc == 3) {
    // ...
} else {
    // error show usage...
}
```

Reply:
used a switch statement instead.